tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -642,14 +642,26 @@ def press_release_volume_up_then_down_n_times(self, n_times):
             };""", script_args=[n_times])
 
     def turn_screen_off(self):
-        self.marionette.execute_script("window.wrappedJSObject.ScreenManager.turnScreenOff(true)")
+        apps = GaiaApps(self.marionette)
+        self.marionette.switch_to_frame()
+        ret = self.marionette.execute_script("window.wrappedJSObject.ScreenManager.turnScreenOff(true)")
+        apps.switch_to_displayed_app()
+        return ret
 
     def turn_screen_on(self):
-        self.marionette.execute_script("window.wrappedJSObject.ScreenManager.turnScreenOn(true)")
+        apps = GaiaApps(self.marionette)
+        self.marionette.switch_to_frame()
+        ret = self.marionette.execute_script("window.wrappedJSObject.ScreenManager.turnScreenOn(true)")
+        apps.switch_to_displayed_app()
+        return ret
 
     @property
     def is_screen_enabled(self):
-        return self.marionette.execute_script('return window.wrappedJSObject.ScreenManager.screenEnabled')
+        apps = GaiaApps(self.marionette)
+        self.marionette.switch_to_frame()
+        ret = self.marionette.execute_script('return window.wrappedJSObject.ScreenManager.screenEnabled')
+        apps.switch_to_displayed_app()
+        return ret
 
```

Comment:
This actually makes us less consistent. In some places we assume that we're in the correct frame, in other places we switch to the system frame but do not attempt to switch back, and here we switch to the system frame and attempt to switch back.

The problem with switching frame is that the test author is not likely to expect this, and may need to account for it. Whilst we can attempt to switch back to the displayed app, this is only an attempt to restore the original frame context - if we were previously in a nested frame, we're going to end up in a different frame. I would favour consistency, and we should either remove the dependency on being in the system frame (where possible), determine if there's a way to somehow fork a process to do the switch and carry out the script, or fall back to some consistent behaviour. Having a decorator for this sounds like a nice approach, and will keep our solution in a central location.

To be clear, I'm r+ this if it works and does not cause regressions, but would like to see a followup that improves our consistency and maintenance.